### PR TITLE
add system client

### PR DESCRIPTION
**Pull request by elenz97**
- add methods providing basic coverage of the `system/gc` API endpoint

### CODE REVIEW

File: system_types.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// AdminJobReq holds request information for admin job
+type AdminJobReq struct {
+	AdminJobSchedule
+	Name       string                 `json:"name"`
+	Status     string                 `json:"status"`
+	ID         int64                  `json:"id"`
+	Parameters map[string]interface{} `json:"parameters"`
+}
+
+// AdminJobSchedule ...
```
Comment (dkorittki) [inline system_types.go:31]: Godoc missing I guess?

File: system_types.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	ScheduleDaily = "Daily"
+	// ScheduleWeekly : 'Weekly'
+	ScheduleWeekly = "Weekly"
+	// ScheduleCustom : 'Custom'
+	ScheduleCustom = "Custom"
+	// ScheduleManual : 'Manual'
+	ScheduleManual = "Manual"
+	// ScheduleNone : 'None'
+	ScheduleNone = "None"
+)
```
Comment (jkmw) [inline system_types.go:20]: maybe use an own type for the consts?

```suggestion
type ScheduleType string
const (
	ScheduleTypeHourly ScheduleType = "Hourly"
	ScheduleTypeDaily               = "Daily"
	ScheduleTypeWeekly              = "Weekly"
	ScheduleTypeCustom              = "Custom"
	ScheduleTypeManual              = "Manual"
	ScheduleTypeNone                = "None"
)
```

File: system_types.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+// AdminJobSchedule ...
+type AdminJobSchedule struct {
+	Schedule *ScheduleParam `json:"schedule"`
+}
+
+// ScheduleParam defines the parameter of schedule trigger
+type ScheduleParam struct {
+	// Daily, Weekly, Custom, Manual, None
+	Type string `json:"type"`
```
Comment (jkmw) [inline system_types.go:39]:
```suggestion
	Type ScheduleType `json:"type"`
```

File: system_types.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package harbor
+
+// To ensure type-safe queries to the harbor API,
+// the following typings include typings from the upstream sources:
+// https://github.com/goharbor/harbor/src/core/api/models
```
[the remark this answers is not in the document]
Comment (elenz97) [reply system_types.go]: All of the files with upstream type definitions now refer to the correct `v1.10.2` branch

**Pull request merged**